app/src/main/java/com/pvp/app/ui/screen/profile/ProfileViewModel.kt

```diff
@@ -1,62 +1,84 @@
 package com.pvp.app.ui.screen.profile
 
+import androidx.compose.ui.graphics.ImageBitmap
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.pvp.app.api.AuthenticationService
 import com.pvp.app.api.UserService
+import com.pvp.app.model.Ingredient
 import com.pvp.app.model.SignOutResult
+import com.pvp.app.model.SportActivity
 import com.pvp.app.model.User
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
+
 @HiltViewModel
 class ProfileViewModel @Inject constructor(
     private val authenticationService: AuthenticationService,
     private val userService: UserService
 ) : ViewModel() {
 
-    private val _user = MutableStateFlow<User?>(null)
-    val user = _user.asStateFlow()
+    private val _state = MutableStateFlow(
+        ProfileState(
+            isLoading = true,
+            user = User(),
+            userAvatar = ImageBitmap(1, 1)
+        )
+    )
+    val state: StateFlow<ProfileState> = _state.asStateFlow()
 
     init {
         viewModelScope.launch {
-            _user.value = userService
+            val user = userService
                 .getCurrent()
                 .first()
-        }
-    }
-
-    fun updateUsername(newUsername: String) {
-        viewModelScope.launch {
-            _user.value?.let {
-                it.username = newUsername
+                ?: return@launch
+            val avatar = userService.resolveAvatar(user.email)
 
-                userService.merge(it)
+            _state.update {
+                ProfileState(
+                    isLoading = false,
+                    user = user,
+                    userAvatar = avatar
+                )
             }
         }
     }
 
-    fun updateMass(newMass: Int) {
+    fun updateUserInformation(
+        newUsername: String? = null,
+        newMass: Int? = null,
+        newHeight: Int? = null,
+        newActivityFilters: List<String>? = null,
+        newIngredientFilters: List<String>? = null
+    ) {
         viewModelScope.launch {
-            _user.value?.let {
-                it.mass = newMass
-
-                userService.merge(it)
+            newUsername?.let {
+                _state.value.user.username = it
             }
-        }
-    }
-
-    fun updateHeight(newHeight: Int) {
-        viewModelScope.launch {
-            _user.value?.let {
-                it.height = newHeight
-
-                userService.merge(it)
+            newMass?.let {
+                _state.value.user.mass = it
+            }
+            newHeight?.let {
```

Comment:
Separate all by single line, those are methods, all looking differently because of referenced variable names. If all were to just be `let` single-line bodies, it could stay grouped. But here that's not the case.